### PR TITLE
Hook for custom parsing of login module options

### PR DESCRIPTION
**Pull request by DoruM**
Hook for custom parsing of login module options

### CODE REVIEW

**Comment by dpwspoon** [comment]
This needs a catchup merge

File: server/src/main/java/org/kaazing/gateway/server/GatewayObserverFactorySpiPrototype.java [hunk 1]
```diff
@@ -92,4 +93,10 @@ public void stoppedGateway(GatewayContext gatewayContext) {
     public void initingGateway(Properties properties, Map<String, Object> injectables) {
         // NOOP: to be extended as desired
     }
+
```
Comment (dpwspoon) [inline server/src/main/java/org/kaazing/gateway/server/GatewayObserverFactorySpiPrototype.java]: Can we move this to a separate API/SPI

**Comment by DoruM** [comment]
Added unit tests.
Moved new function in new interface.

**Comment by dpwspoon** [comment]
LGTM

**Comment by dpwspoon** [comment]
Looks excellent to me, but some files are missing license headers which is failing travis :-(.  

```
Failed to execute goal com.mycila:license-maven-plugin:2.11:check (default) on project gateway.server: Some files do not have the expected license header -> [Help 1]
```

**Pull request merged**